internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -0,0 +1,572 @@
+				Properties: &mongoclusters.MongoClusterProperties{},
+			}
+
+			if state.AdministratorUserName != "" {
+				parameter.Properties.Administrator = &mongoclusters.AdministratorProperties{
+					UserName: pointer.To(state.AdministratorUserName),
+					Password: pointer.To(state.AdministratorPassword),
+				}
+			}
+
+			if state.CreateMode != "" {
+				parameter.Properties.CreateMode = pointer.To(mongoclusters.CreateMode(state.CreateMode))
+			}
+
+			parameter.Properties.PreviewFeatures = expandPreviewFeatures(state.PreviewFeatures)
+
+			if state.ShardCount != 0 {
+				parameter.Properties.Sharding = &mongoclusters.ShardingProperties{
+					ShardCount: pointer.To(state.ShardCount),
+				}
+			}
+
+			if state.CreateMode == string(mongoclusters.CreateModeGeoReplica) {
+				parameter.Properties.ReplicaParameters = &mongoclusters.MongoClusterReplicaParameters{
+					SourceLocation:   state.SourceLocation,
+					SourceResourceId: state.SourceServerId,
+				}
+			}
+
+			if state.ComputeTier != "" {
+				parameter.Properties.Compute = &mongoclusters.ComputeProperties{
+					Tier: pointer.To(state.ComputeTier),
+				}
+			}
+
+			if state.HighAvailabilityMode != "" {
+				parameter.Properties.HighAvailability = &mongoclusters.HighAvailabilityProperties{
+					TargetMode: pointer.To(mongoclusters.HighAvailabilityMode(state.HighAvailabilityMode)),
+				}
+			}
+
+			if state.PublicNetworkAccessEnabled {
+				parameter.Properties.PublicNetworkAccess = pointer.To(mongoclusters.PublicNetworkAccessEnabled)
+			} else {
+				parameter.Properties.PublicNetworkAccess = pointer.To(mongoclusters.PublicNetworkAccessDisabled)
+			}
+
+			if state.StorageSizeInGb != 0 {
+				parameter.Properties.Storage = &mongoclusters.StorageProperties{
+					SizeGb: pointer.To(state.StorageSizeInGb),
+				}
+			}
+
+			if state.Tags != nil {
+				parameter.Tags = pointer.To(state.Tags)
+			}
+
+			if state.Version != "" {
+				parameter.Properties.ServerVersion = pointer.To(state.Version)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, parameter); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r MongoClusterResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MongoCluster.MongoClustersClient
+
+			id, err := mongoclusters.ParseMongoClusterID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Info("Decoding state...")
+			var state MongoClusterResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			model := existing.Model
+			if model == nil {
+				return fmt.Errorf("reading %s for update: model was nil", *id)
+			}
```

Comment:
For consistency update this to 
```suggestion
	if existing.Model == nil {
		return fmt.Errorf("retrieving %s: `model` was nil", *id)
	}
	if existing.Model.Properties == nil {
		return fmt.Errorf("retrieving %s: `properties` was nil", *id)
	}
	payload := existing.Model
```

Reply:
Fixed.